broadcaster/_backends/pulsar.py

```diff
@@ -14,64 +15,80 @@ def __init__(self, url: str):
         self._port = parsed_url.port or 6650
         self._service_url = f"pulsar://{self._host}:{self._port}"
         self._client = None
-        self._producer = None
-        self._consumer = None
+        self._producers = {}
+        self._consumers = {}
+        self._subscribed_channels = set()
```

Comment:
Isn't it exactly the same as `self._consumers.keys()`? Do we really need to store it twice?